src/tools/miri/src/alloc_addresses/mod.rs

```diff
@@ -318,6 +351,21 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
         Ok(base_ptr.wrapping_offset(offset, ecx))
     }
 
+    // This returns some prepared `MiriAllocBytes`, either because `addr_from_alloc_id` reserved
+    // memory space in the past, or by doing the pre-allocation right upon being called.
+    fn take_prepared_alloc_bytes(&self, id: AllocId, kind: MemoryKind) -> InterpResult<'tcx, MiriAllocBytes> {
+        let ecx = self.eval_context_ref();
+        // This additional call ensures that some `MiriAllocBytes` are prepared.
+        ecx.addr_from_alloc_id(id, kind)?;
+        // TODO: Better debug message?
+        let mut global_state = ecx.machine.alloc_addresses.borrow_mut();
+        let prepared_alloc_bytes = global_state
+            .prepared_alloc_bytes
+            .remove(&id)
+            .expect("alloc bytes for alloc_id could not prepared");
```

Comment:
```suggestion
            .expect("alloc bytes for {id:?} has not been prepared");
```

Reply:
Oh, not sure how important this is, but `expect` does not automatically format the `".. {id:?}"` inside the string. Using `format!` would seem expensive (`String`), so do I:
1. leave away the `{id:?}`,
2. use something like `.unwrap_or_else(|| panic!(".. {id:?}"))`,
3. (some other option)?

Reply:
Ah right, good point. `unwrap_or_else(|| panic!(".. {id:?}"))` seems good. Or just make it `.expect("alloc bytes have not been prepared")`. I was mostly confused why it said `for alloc_id`.